src/builder.rs

```diff
@@ -34,12 +34,24 @@ const MIN_SHIELDED_OUTPUTS: usize = 2;
 pub enum BundleType {
     /// A transactional bundle will be padded if necessary to contain at least 2 outputs,
     /// irrespective of whether any genuine outputs are required.
-    Transactional { anchor: Node },
+    Transactional {
+        /// A flag that, when set to `true`, indicates that the resulting bundle should be produced
+        /// with the minimum required outputs irrespective of whether any outputs have been
+        /// requested; if no explicit outputs have been added, all of the outputs in the resulting
+        /// bundle will be dummies.
+        outputs_required: bool,
```

Comment:
This flag is useless compared to the `bundle_required` flag in https://github.com/zcash/orchard/pull/409, because if there are no dummy Spends then the only source of balancing funds for these required outputs is the `valueBalance`, which immediately leaks whether these outputs are dummies or not.

I think instead this should be `bundle_required`, and it forces the inclusion of 1 spend and 2 outputs (i.e. the minimum non-zero number of spends and outputs that regular Sapling-only transactions will have after padding is applied).